lib/shared/puppet_forge/unpacker.rb

```diff
@@ -0,0 +1,78 @@
+require 'pathname'
+require 'tmpdir'
+require 'shared/puppet_forge/error'
+require 'shared/puppet_forge/tar'
+
+module PuppetForge
+  class Unpacker
+    # Unpack a tar file into a specified directory
+    #
+    # @param filename [String] the file to unpack
+    # @param target [String] the target directory to unpack into
+    def self.unpack(filename, target, tmpdir)
+      inst = self.new(filename, target, tmpdir)
+      inst.unpack
+      inst.move_into(target)
+    end
+
+    # Set the owner/group of the target directory to those of the source
+    #
+    # @param source [Pathname] source of the permissions
+    # @param target [Pathname] target of the permissions change
+    # @param microsoft_windows boolean flag for whether this system is windows or not
+    def self.harmonize_ownership(source, target, microsoft_windows = false)
+      if !microsoft_windows
+        FileUtils.chown_R(source.stat.uid, source.stat.gid, target)
+      end
+    end
+
+    # @param filename [String] the file to unpack
+    # @param target [String] the target directory to unpack into
+    def initialize(filename, target, tmpdir)
+      @filename = Pathname.new(filename)
```

Comment:
We convert `filename` to a `Pathname` here, and on line 40 we call `@filename.to_s` which means we're round tripping through Pathname to a string. Normally I really prefer using Pathname objects over Strings because it's easier to test and stub, but it doesn't look like we're actually taking advantage of any functionality on Pathname. Does it make sense to do this for consistency or some other reason?